pkg/harvester/edit/kubevirt.io.virtualmachine/VirtualMachinePciDevices/index.vue

```diff
@@ -139,11 +182,14 @@ export default {
     deviceOpts() {
       return Object.keys(this.uniqueDevices).map((deviceId) => {
         const device = this.uniqueDevices[deviceId].deviceCRDs[0];
+        const numberOfDeviceEnabled = this.uniqueDevices[deviceId].deviceCRDs.length;
+        const numberOfDeviceInUse = this.devicesInUse[device?.status?.resourceName]?.count || 0;
 
         return {
           resourceName: device?.status?.resourceName,
           value:        deviceId,
-          label:        deviceId
+          label:        deviceId,
+          disabled:     numberOfDeviceInUse === numberOfDeviceEnabled
```

Comment:
Why is `deviceCRDs.length === resourceName.count` used here to determine whether to disabled.

The two pci's I have used in the vm, but when I create a new vm again, one of them is still selected and the other one is not selected because the count is exactly equal to 1 .  `deviceCRDs.length === 1,  count ===1`
<img width="1700" alt="image" src="https://user-images.githubusercontent.com/24985926/196895928-cf31d790-7ff6-4c9f-a5f5-2e49b0ed0bb5.png">

Reply:
Thank you for catching that. I made a mistake calculating `resourceName.count` : it should be the number of times this device is used in other VMs. `deviceCRDs` is an array of all of the enabled instances of a device in the cluster. So if `deviceCRDs.length===count` then all instances of the device have been used by other VMs, and it cannot be added to this VM. I added some comments in code to maybe help clarify, too.